### ISSUE TITLE
Bug with setup script: unsuccessful instalation of "kubeadm"

### ISSUE THREAD

**Issue by EstellaPaula** [vhive-serverless/vHive#606]
**What I did:**
When running the command from quickstart guide, step II.4.1, I noticed that the command **"kubeadm"** throws "command not found" prompt. To double check, I searched with "where" command to see if the binary of "kubeadm" can be found, but it seemed that the command was not installed at all. I then installed "kubeadm" command manually using the instructions from [here](http://pwittrock.github.io/docs/setup/independent/install-kubeadm/#installing-kubelet-and-kubeadm) and retried the command from step II.4.1. However, the node failed to join the cluster prompting "prefetch failed" and the cause that I could understand from the output of the command was from failed prefetching of the ConfigMap or something similar.

**To Reproduce**
Follow the steps in the quickstart guide on a fresh installation on CloudLab with the recommended profile from the guide.

**What I would expect:**
I expected the command to run successfully and the worker node to join the cluster, with output as mentioned in the guide:

```
This node has joined the cluster:
	* Certificate signing request was sent to apiserver and a response was received.
	* The Kubelet was informed of the new secure connection details.
```

**Necessary logs:**
I do not have the logs anymore for the config map issue, as I have reloaded the machines and found a solution, but I am pretty confident I can reproduce the bug if needed, hence why I am posting this here and not on GH first.
I ran the quickstart guide commands on fresh installations 2 different times (reloaded the nodes in between) and this seems to be reproduceable (I used the suggested CloudLab profile though, haven't tried different machines/distros). Although I got no errors in previous parts of the guide.

**Identified issue:**
It appears that the setup script doesn't install the "kubeadm" command because the environment variable K8S\_VERSION is set to empty\_string. I am not quite sure why that is because I inspected the ../main/scripts/install\_stock.sh file, and it appears that the variable gets set at line 48, right before using it to install "kubeadm". But if I echo the environment variable after running the script I get empty string.

I solved the issue by inspecting the installed version of kubernetes with:
`kubectl version`
and setting the variable by hand to the current version used by kubernetes on the machine with waht was in my case version "v1.23.5":
`export K8S_VERSION=1.23.5*`
Then I reran by hand the command from the script that ensured "kubeadm" installation:
`sudo apt-get -y install cri-tools ebtables ethtool kubeadm=$K8S_VERSION kubectl=$K8S_VERSION kubelet=$K8S_VERSION kubernetes-cni >> /dev/null`

After this, I reran the command from the quickstart quide, step II.4.1, and the node joined the cluster successfully.

**Comment by ustiugov** [comment]
@EstellaPaula is the problem still relevant or you found a fix?

**Comment by EstellaPaula** [comment]
> @EstellaPaula is the problem still relevant or you found a fix?

I have encountered a similar issue with "kubectl" not being installed in the same spot as described here. I suspect that it came from the same problem. I have been able to reproduce this issue only on multi node clusters where only some of the nodes exhibit this, if at all. I usually bypass it by reloading the node and redoing the steps from the quickstart guide, just to make sure everything is installed properly by the setup scripts and there are no other dependencies that I miss by installing kube by hand. Worth mentioning that I spent the past weeks working with an older version of the repo, so I am unsure if this still occurs on the main branch.
